src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp

```diff
@@ -147,11 +147,19 @@ void CachedOnDiskReadBufferFromFile::initialize(size_t offset, size_t size)
 }
 
 CachedOnDiskReadBufferFromFile::ImplementationBufferPtr
-CachedOnDiskReadBufferFromFile::getCacheReadBuffer(const FileSegment & file_segment) const
+CachedOnDiskReadBufferFromFile::getCacheReadBuffer(const FileSegment & file_segment)
 {
     ProfileEventTimeIncrement<Microseconds> watch(ProfileEvents::CachedReadBufferCreateBufferMicroseconds);
 
     auto path = file_segment.getPathInLocalCache();
+    if (cache_file_reader)
+    {
+        chassert(cache_file_reader->getFileName() == path);
+        if (cache_file_reader->getFileName() == path)
+            return cache_file_reader;
+
+        cache_file_reader.reset();
```

Comment:
Regarding resetting `cache_file_reader`, as discussed, it could be nice to reuse already allocated buffer but reset only other members

Reply:
When discussing it yesterday I forgot that we do not allocate it already and reuse, so no need to do anything here anymore.